backend/src/main/java/com/example/user/UserController.java

```diff
@@ -35,6 +31,27 @@ public Optional<User> findById(@PathVariable int id) {
         return userRepository.findById(id);
     }
 
+    @PostMapping("/login")
+    public String loginUser(@RequestBody User user) {
+        // Check if user with specified name or email exists
+        if (!userRepository.existsByName(user.getName())) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "No user with specified name found");
+        }
+
+        User specifiedUser = userRepository.findByName(user.getName());
+
+        if (!specifiedUser.getPassword().equals(user.getPassword())) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Wrong password");
+        }
```

Comment:
W momencie, w którym piszesz taki komentarz w metodzie/funkcji zawsze trzeba się zastanowić czy to nie powinno być wydzielone do oddzielnej metody. Wtedy nazwa metody jest komentarzem samym w sobie. W tym przypadku te sprawdzenie powinno się odbywać w `UserService` - którego na ten moment nie masz niestety. Warto by go było teraz utworzyć. `Controller` nie powinien bezpośrednio korzystać z `Repository` - zawsze powinno się to odbywać za pomocą serwisu. Rozwiązuje to dużo problemów z łapaniem różnych błędy. Odciąża kontroler od odpowiedzialności za to

Reply:
W dużym skrócie jest to opisane tutaj:
https://www.baeldung.com/spring-component-repository-service

Reply:
Dodaj to do funkcji validateUser() lub coś w tym stylu.